clients/src/main/java/org/apache/kafka/common/serialization/Serdes.java

```diff
@@ -0,0 +1,146 @@
+
+/**
+ * Factory for creating serializers / deserializers.
+ */
+public class SerDes {
+
+    static private final Serializer<Long> longSerializer = new LongSerializer();
+    static private final Deserializer<Long> longDeserializer = new LongDeserializer();
+
+    static private final Serializer<Integer> integerSerializer = new IntegerSerializer();
+    static private final Deserializer<Integer> integerDeserializer = new IntegerDeserializer();
+
+    static private final Serializer<String> stringSerializer = new StringSerializer();
+    static private final Deserializer<String> stringDeserializer = new StringDeserializer();
+
+    static private final Serializer<byte[]> bytesSerializer = new ByteArraySerializer();
+    static private final Deserializer<byte[]> bytesDeserializer = new ByteArrayDeserializer();
+
+
+    static public final class LongSerDe implements SerDe<Long> {
+
+        @Override
+        public Serializer<Long> serializer() {
+            return longSerializer;
+        }
+
+        @Override
+        public Deserializer<Long> deserializer() {
+            return longDeserializer;
+        }
+    }
+
+    static public final class IntegerSerDe implements SerDe<Integer> {
+        @Override
+        public Serializer<Integer> serializer() {
+            return integerSerializer;
+        }
+
+        @Override
+        public Deserializer<Integer> deserializer() {
+            return integerDeserializer;
+        }
+    }
+
+    static public final class StringSerDe implements SerDe<String> {
+        @Override
+        public Serializer<String> serializer() {
+            return stringSerializer;
+        }
+
+        @Override
+        public Deserializer<String> deserializer() {
+            return stringDeserializer;
+        }
+    }
+
+    static public final class ByteArraySerDe implements SerDe<byte[]> {
+        @Override
+        public Serializer<byte[]> serializer() {
+            return bytesSerializer;
+        }
+
+        @Override
+        public Deserializer<byte[]> deserializer() {
+            return bytesDeserializer;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    static public <T> SerDe<T> serialization(Class<T> type) {
+        if (String.class.isAssignableFrom(type)) {
+            return (SerDe<T>) STRING();
+        }
+
+        if (Integer.class.isAssignableFrom(type)) {
+            return (SerDe<T>) INTEGER();
+        }
+
+        if (Long.class.isAssignableFrom(type)) {
+            return (SerDe<T>) LONG();
+        }
+
+        if (byte[].class.isAssignableFrom(type)) {
+            return (SerDe<T>) BYTE_ARRAY();
+        }
+
+        // TODO: we can also serializes objects of type T using generic Java serialization by default
+        throw new IllegalArgumentException("Unknown class for built-in serializer");
+    }
+
+    static public <T> SerDe<T> serialization(final Serializer<T> serializer, final Deserializer<T> deserializer) {
```

Comment:
I think we should rename this method name, because `serialization` (to me) means the Serializer functionality *only*, at least in the way we have decided to talk about it in the code.

I'd suggest to rename `serialization(...)` to e.g. `serdeFrom(...)`.